### PR TITLE
Extract patterns from output with `extract_match`

### PR DESCRIPTION
**Pull request by beccasaurus**
```yaml
extract_match:
  pattern: projects/(.*)
  variable: project_id

extract_match:
  pattern: projects/(.*)/companies/(.*)
  groups:
  - project_id
  - company_id
```

This is the last blocker for my to use sample-tester with my samples.

### CODE REVIEW

File: tests/testdata/caserunner_test.yaml [hunk 1]
```diff
@@ -182,6 +182,78 @@ test:
...
+        - /bin/echo "It was the best of times"
+      - extract_match:
+          pattern: ^([^\s]+) was (.*) of
+          variable: my_variable
+          groups:
+          - first_capture
+          - second_capture
+          - third_capture # not matched
+      - assert_success:
+        - if extract_match fails, this is not run
```
Comment (vchudnov-g) [inline tests/testdata/caserunner_test.yaml]: nit: final newline

File: sampletester/caserunner.py [hunk 2]
```diff
@@ -19,6 +19,7 @@
 import subprocess
 import traceback
 import uuid
+import re
```
Comment (vchudnov-g) [inline sampletester/caserunner.py]: nit: put in alphabetical order

File: sampletester/caserunner.py [hunk 3]
```diff
@@ -86,6 +87,8 @@ def __init__(self, environment: testenv.Base,
         # the test to continue even if an expectation is not met).
         # "expect_contains": (self.expect_contains, self.params_for_contains),
         # "expect_not_contains": (self.expect_not_contains, self.params_for_contains),
+
+        "extract_match": (self.extract_match, self.yaml_extract_match),
```
Comment (vchudnov-g) [inline sampletester/caserunner.py]: move into the "Other functions" section

File: sampletester/caserunner.py [hunk 4]
```diff
@@ -160,6 +163,42 @@ def yaml_get_env(self, parts):
...
+          if len(captures) > idx:
+            self.local_symbols[variable_name] = captures[idx]
+    return None, None
+
+  # Extracts regular expression captures from output via code.
+  def yaml_extract_match(self, parts):
+    key_pattern = 'pattern'
+    key_variable = 'variable'
+    key_groups = 'groups'
+    self.extract_match(parts.get(key_pattern), parts.get(key_variable),
```
[the remark this answers is not in the document]
Comment (vchudnov-g) [reply sampletester/caserunner.py]: (or, obvs, a list comprehension of the above)

**Comment by beccasaurus** [comment]
Pushed changes!

**Pull request merged**